tests/test_hand.py

```diff
@@ -3,17 +3,28 @@
 import pytest
 from osgeo import gdal, ogr
 from osgeo_utils.gdalcompare import find_diff
+import numpy as np
 
 from asf_tools import hand
 
 HAND_BASINS = '/vsicurl/https://hyp3-testing.s3-us-west-2.amazonaws.com/' \
-              'asf-tools/hand/hybas_af_lev12_v1c_firstpoly.geojson'
+              'asf-tools/S1A_IW_20230228T120437_DVR_RTC30/hand/hybas_af_lev12_v1c_firstpoly.geojson'
 GOLDEN_HAND = '/vsicurl/https://hyp3-testing.s3-us-west-2.amazonaws.com/' \
-              'asf-tools/hand/hybas_af_lev12_v1c_firstpoly.tif'
+              'asf-tools/S1A_IW_20230228T120437_DVR_RTC30/hand/hybas_af_lev12_v1c_firstpoly.tif'
 
 gdal.UseExceptions()
 
 
+def nodata_equal_nan(GOLDEN_HAND, out_hand):
+    ds_golden = gdal.Open(str(GOLDEN_HAND))
+    ds_out = gdal.Open(str(out_hand))
+    nodata_golden = ds_golden.GetRasterBand(1).GetNoDataValue()
+    nodata_out = ds_out.GetRasterBand(1).GetNoDataValue()
+    if nodata_golden and nodata_out:
+        return np.isnan(nodata_golden) and np.isnan(nodata_out)
```

Comment:
I also am confused why this is needed...

Reply:
if both files have the nodata value = numpy.nan, it is supposed that two nodata is the same, but gdalcompare.py has a bug. it treats these two nodata values as a different.  Because two variables with numpy.nan returns False if you use == to compare them. a1=np.nan, b1=np.nan, a1 == b1 returns False. I use np.isnan() to judge if both nodata are np.nan, we allow the assert diff_count ==1.